cdap-ui/app/cdap/components/AbstractWidget/WranglerEditor/index.tsx

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+import PropTypes from 'prop-types';
+import * as React from 'react';
+import { withStyles, WithStyles } from '@material-ui/styles';
+import CodeEditor from 'components/CodeEditor';
+import Button from '@material-ui/core/Button';
+import ThemeWrapper from 'components/ThemeWrapper';
+import If from 'components/If';
+import { Modal, ModalBody } from 'reactstrap';
+import DataPrepHome from 'components/DataPrepHome';
+import { preventPropagation } from 'services/helpers';
+import LoadingSVG from 'components/LoadingSVG';
+// This artifact will stay until we migrate dataprep to use css-in-js
+require('./wrangler-modal.scss');
+
+const styles = (theme) => {
+  return {
+    root: {
+      width: '100%',
+    },
+    wrangleButton: {
+      margin: '0 10px',
+    },
+    modalBtnClose: {
+      height: '50px',
+      width: '50px',
+      boxShadow: 'none',
+      border: 0,
+      background: 'transparent',
+      borderLeft: `1px solid ${theme.palette.grey['300']}`,
+      fontWeight: 'bold' as 'bold',
+      fontSize: '1.5rem',
+      '&:hover': {
+        background: theme.palette.blue['40'],
+        color: 'white',
+      },
+    },
+  };
+};
+
+interface IWranglerEditorWidgetAttributes {
+  rows?: string;
+  placeholder?: string;
+}
+
+interface IWranglerEditorWidgetConfig {
+  name: string;
+  label: string;
+  'widget-type': string;
+  'widget-attributes': IWranglerEditorWidgetAttributes;
+}
+
+// This is not ideal to hardcode wrangler plugin properties here.
+// but we are modifying these properties and if that changes then this
+// type has to change.
+interface IWranglerEditorProperties {
+  workspaceId: string;
+  schema: string;
+  directives: string;
+}
+
+interface IWranglerEditorProps extends WithStyles<typeof styles> {
+  value: string;
+  config: IWranglerEditorWidgetConfig;
```

Comment:
pass in the specific props instead of the whole config?

Reply:
Any specific reason? The `IWranglerEditorWidgetConfig` represents the config that will be passed on from the widget json to the individual components. If we make them individual ones then the configuration-groups component that we will implement will need to know what properties individual components need to be passed on (essentially the factory). If for this contract it will be,
```{  value, config, properties, onChange, disabled } ```
which can be standardized across all components. May be we can move this config type to the configuration-groups and have that be reused across all the widgets.

Reply:
talked with @ajainarayanan, will keep this pattern. All the other widgets will be migrated to this format on the ConfigurationGroup component work